feat(retry): add configurable delay and retry limits

dannycreations opened the pull request:
## Implementation

- Implement retry logic in Task core to respect new configurations
- Update settings UI with sliders for delay and retry count
- Add localization for new settings across all supported languages
- Add unit tests for auto-retry logic

## Screenshots

<img width="726" height="187" alt="image" src="https://github.com/user-attachments/assets/77a27f93-308d-43de-b71a-02b9f3743e1d" />
<img width="833" height="162" alt="image" src="https://github.com/user-attachments/assets/523a894e-f910-4523-8bfe-0d2db4d43f4a" />

marius-kilocode commented on cli/src/config/mapper.ts:
```diff
@@ -61,6 +61,8 @@ export function mapConfigToExtensionState(
 			alwaysAllowBrowser: autoApprovalEnabled && (autoApproval?.browser?.enabled ?? false),
```
Changeset missing

marius-kilocode commented on cli/src/host/ExtensionHost.ts:
```diff
@@ -1002,6 +1002,12 @@ export class ExtensionHost extends EventEmitter {
 		if (configState.requestDelaySeconds !== undefined) {
 			autoApprovalSettings.requestDelaySeconds = configState.requestDelaySeconds
 		}
+		if (configState.requestRetryMax !== undefined) {
```
Did you also test if the CLI is working? How would CLI users configure this? Either we need some docs here or maybe a CLI option.

dannycreations commented on src/core/task/Task.ts:
```diff
@@ -4472,16 +4483,28 @@ export class Task extends EventEmitter<TaskEvents> implements TaskLike {
+				case "exponential":
+				default:
+					delaySeconds = Math.ceil(baseDelay * Math.pow(2, retryAttempt))
+					break
```
hmm i just need the `constant` delay, the default was `exponential`, so i can reduce to 2 constant and exponential or constant and linear which one did you prefer? or default to constant since now user can configure delay? i thought more user options is good ux but you point out maintain cost which is valid also

dannycreations committed:
feat(retry): implement configurable delay and max retries

- Add `alwaysApproveResubmit`, `requestDelaySeconds`, and `requestRetryMax` to global settings and state.
- Update `Task` logic to respect maximum retry limits and use configurable delays instead of hardcoded exponential backoff.
- Integrate retry configuration sliders and toggles into the Auto-Approve settings UI.
- Add comprehensive unit tests for auto-retry logic, delay calculations, and state constraints.
- Update localization files for all supported languages with new retry-related labels and descriptions.

dannycreations committed:
Update webview-ui/src/components/settings/AutoApproveSettings.tsx

Co-authored-by: kiloconnect[bot] <240665456+kiloconnect[bot]@users.noreply.github.com>

dannycreations committed:
Update webview-ui/src/i18n/locales/en/settings.json

Co-authored-by: kiloconnect[bot] <240665456+kiloconnect[bot]@users.noreply.github.com>

dannycreations committed:
Update webview-ui/src/components/settings/SettingsView.tsx

Co-authored-by: kiloconnect[bot] <240665456+kiloconnect[bot]@users.noreply.github.com>

dannycreations committed:
Update packages/types/src/global-settings.ts

Co-authored-by: kiloconnect[bot] <240665456+kiloconnect[bot]@users.noreply.github.com>

dannycreations committed:
Update webview-ui/src/components/settings/AutoApproveSettings.tsx

Co-authored-by: kiloconnect[bot] <240665456+kiloconnect[bot]@users.noreply.github.com>
